app/components/WalletProvider.tsx

```diff
@@ -1,43 +1,32 @@
 import type { ReactElement } from "react";
 import { ClientOnly } from "remix-utils";
-import {
-  WagmiConfig,
-  createClient,
-  configureChains,
-  chain,
-  defaultChains,
-  defaultL2Chains,
-} from "wagmi";
+import { WagmiConfig, createClient, configureChains, chain, defaultChains, defaultL2Chains } from "wagmi";
 import { publicProvider } from "wagmi/providers/public";
 // import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet'
 import { InjectedConnector } from "wagmi/connectors/injected";
 import { MetaMaskConnector } from "wagmi/connectors/metaMask";
 import { WalletConnectConnector } from "wagmi/connectors/walletConnect";
 import { infuraProvider } from "wagmi/providers/infura";
 
-export default function WalletProvider({
-  children,
-  network,
-}: {
-  children: ReactElement;
-  network: string;
-}) {
+export default function WalletProvider({ children, network }: { children: ReactElement; network: string }) {
   let configureChainObj;
 
   //TODO create config of all chains we support
 
   // TODO configure for POLYGON
 
-  if (network === "ropsten") {
-    const infuraId = "54fcc811bac44f99b84a04a4a3e2f998";
-    configureChainObj = configureChains(defaultChains, [
-      infuraProvider({ infuraId }),
-      publicProvider(),
-    ]);
+  const infuraId = "54fcc811bac44f99b84a04a4a3e2f998";
+
+  if (network === "ropsten" || network == "goerli") {
+    configureChainObj = configureChains(
+      [chain.ropsten, chain.goerli],
+      [infuraProvider({ infuraId }), publicProvider()]
+    );
+  } else if (network == "polygon") {
+    configureChainObj = configureChains([chain.polygon], [infuraProvider({ infuraId }), publicProvider()]);
   } else {
-    const infuraId = "54fcc811bac44f99b84a04a4a3e2f998";
     configureChainObj = configureChains(
-      [chain.polygon],
+      [chain.localhost, chain.hardhat],
```

Comment:
@jdaly13 I saw your comment on the previous draft of this PR - and added both chains to be supported with this configuration.

Right now all chains are using the same config so we might even be able to squash this and remove the if/else block completely - but tbh I'm a bit surprised it does work using the same infuraID for testnets and mainnets (polygon)

Reply:
well the api key I believe is the same for all EVM protocols also if it can't find Infura I believe publicProvider is your own node running on localhost